src/sentry/snuba/metrics.py

```diff
@@ -1229,15 +1235,19 @@ def get_series(self, projects: Sequence[Project], query: QueryDefinition) -> dic
                         results[entity]["totals"]["data"] += snuba_query_data_dict[group_tuple]
         else:
             snuba_queries = SnubaQueryBuilder(projects, query).get_snuba_queries()
-            results = {
-                entity: {
-                    # TODO: Should we use cache?
-                    key: raw_snql_query(query, use_cache=False, referrer=f"api.metrics.{key}")
-                    for key, query in queries.items()
-                    if query is not None
-                }
-                for entity, queries in snuba_queries.items()
-            }
+            results = {}
+            for entity, queries in snuba_queries.items():
+                results.setdefault(entity, {})
+                for key, snuba_query in queries.items():
+                    if snuba_query is None:
+                        continue
+
+                    snuba_query = snuba_query.set_limit(limit)
+                    snuba_query = snuba_query.set_offset(offset)
```

Comment:
Here it gets tricky:

1. The offset and limit should be multiplied by the number of metrics queried from the current entity.
2. For time series, we would even need to multiply again by the number of time slots.

For simplicity's sake, we might want to ignore pagination args when there is no `orderBy` clause (i.e. when there are time series involved).